src/utils/storage.ts

```diff
@@ -0,0 +1,319 @@
+  }
+
+  clearUserInfo() {
+    this.delete(StorageKey.UserInfo)
+  }
+
+  setIsLoggedIn() {
+    this.set(StorageKey.IsLoggedIn, true)
+  }
+
+  getIsLoggedIn() {
+    return this.get<boolean>(StorageKey.IsLoggedIn) ?? false
+  }
+
+  clearIsLoggedIn() {
+    this.delete(StorageKey.IsLoggedIn)
+  }
+
+  setSession(val: { sessionID: string; timestamp: number }) {
+    this.set(StorageKey.Session, val)
+  }
+
+  getSession() {
+    return this.get<{ sessionID: string; timestamp: number }>(
+      StorageKey.Session
+    )
+  }
+
+  clearSession() {
+    this.delete(StorageKey.Session)
+  }
+
+  setHasMFA(id: string) {
+    this.set(`${id}-${StorageKey.HasMFA}`, '1')
+  }
+
+  getHasMFA(id: string) {
+    return this.storage.get<string>(`${id}-${StorageKey.HasMFA}`) == '1'
+  }
+
+  setLoginSrc(src: string) {
+    this.set(StorageKey.LoginSrc, src)
+  }
+
+  getLoginSrc() {
+    return this.get<string>(StorageKey.LoginSrc)
+  }
+
+  clearLoginSrc() {
+    return this.delete(StorageKey.LoginSrc)
+  }
+
+  setLoginCount(userId: string, val: number) {
+    return this.set(`${userId}-${StorageKey.LoginCount}`, val)
+  }
+
+  incrementLoginCount(userId: string) {
+    let loginCount = this.getLoginCount(userId)
+    loginCount = loginCount ? Number(loginCount) + 1 : 1
+    this.setLoginCount(userId, loginCount)
+  }
+
+  getLoginCount(userId: string) {
+    return this.get<number>(`${userId}-${StorageKey.LoginCount}`)
+  }
+
+  setMFASkip(userId: string, val: number) {
+    return this.set(`${userId}-${StorageKey.SkipMFAUntil}`, val)
+  }
+
+  getMFASkip(userId: string) {
+    return this.get<number>(`${userId}-${StorageKey.SkipMFAUntil}`)
+  }
+
+  setMFADND(userId: string) {
+    return this.set(`${userId}-${StorageKey.MFADND}`, '1')
+  }
+
+  HasMFADND(userId: string) {
+    return this.get<string>(`${userId}-${StorageKey.MFADND}`) == '1'
+  }
+
+  setNFTList(address: string, items: NFTItem[]) {
+    this.set(`${address}-${StorageKey.NFT}`, items)
+  }
+
+  getNFTList(address: string): NFTItem[] {
+    return this.get(`${address}-${StorageKey.NFT}`) ?? []
+  }
+
+  setAssetContractList(address: string, chainId: number, val: AssetContract[]) {
+    this.set(`${address}/${chainId}/${StorageKey.AssetContract}`, val)
+  }
+
+  getAssetContractList(address: string, chainId: number) {
+    return (
+      this.get<AssetContract[]>(
+        `${address}/${chainId}/${StorageKey.AssetContract}`
+      ) ?? []
+    )
+  }
+
+  setPK(val: { pk: string; id: string; exp: dayjs.Dayjs | undefined }) {
+    this.set(StorageKey.PK, val)
+  }
+
+  getPK() {
+    return this.get<{
+      pk: string
+      id: string
+      exp: dayjs.Dayjs | undefined
+    }>(StorageKey.PK)
+  }
+
+  clearPK() {
+    this.delete(StorageKey.PK)
+  }
+}
+
+interface IStorage {
+  set(key: string, value: unknown): void
+  get<T>(key: string): T | null
+  delete(key: string): void
+}
+
+class InMemoryStorage implements IStorage {
+  private map = {}
```

Comment:
Simply storing things as properties of an object may cause things like prototype pollution and cause vulnerabilities. I don't know if we store untrusted keys but using a `Map` would be safer here.

Reply:
@mmjee updated